File: Baekjun/10.4/9935. 문자열 폭발/solution.py
# BOJ 9935. 문자열 폭발


def solution(string, bomb):
    stack = []
    for s in string:
        stack.append(s)

        if s == bomb[-1] and "".join(stack[-len(bomb) :]) == bomb:
            del stack[-len(bomb) :]

    return "".join(stack) if stack else "FRULA"


if __name__ == "__main__":
    string = input()
    bomb = input()
    print(solution(string, bomb))

File: Baekjun/10.4/9935. 문자열 폭발/solution.swift
func solution(_ string: String, _ bomb: String) -> String {
    var stack = [Character]()

    for s in string {
        stack.append(s)

        if s == bomb.last, String(stack.suffix(bomb.count)) == bomb {
            stack.removeLast(bomb.count)
        }
    }

    return stack.isEmpty ? "FRULA" : String(stack)
}

if let string = readLine(), let bomb = readLine() {
    print(solution(string, bomb))
}
